src/main/java/be/acara/events/controller/EventController.java

```diff
@@ -45,13 +44,16 @@ public ResponseEntity deleteEvent(@PathVariable("id") long id) {
         return ResponseEntity.ok(eventService.getAllCategories());
     }
 
-    @PostMapping
+    @PostMapping("/new-event")
     public ResponseEntity<EventDto> addEvent(@RequestBody @Valid EventDto event) {
-        if (event.getId() != null) {
-            throw new IdAlreadyExistsException("A new entity cannot already contain an id");
-        }
         EventDto eventDto = eventService.addEvent(event);
-        URI uri = URI.create(String.format("/api/events/%d", eventDto.getId()));
+        URI uri = URI.create("/api/events/new-event");
         return ResponseEntity.created(uri).body(eventDto);
     }
+
+    @PostMapping("/edit-event/{id}")
```

Comment:
Editing is either put or patch, not post. The edit-event part is unnecessary as well.